### ISSUE TITLE
loss nan

### ISSUE THREAD

**Issue by wangwuwu** [Nu1sance/MessDet#3]
hi, where i set amp=true ,the loss is nan

**Comment by Nu1sance** [comment]
Thanks for your report.
We did not enable AMP during training in our experiments, and the current implementation is primarily trained and tested under FP32 precision.
Since E2CNN is an older library, it may not support AMP well. We will consider finding out the reasons and improving them.

**Comment by wangwuwu** [comment]
thank you for your reply， i  also want train a small model by changing deepen_factor = 0.33,widen_factor = 0.5, but  ‘assert input.type == self.in_type’  in R2Upsampling.forward throwing an error

**Comment by Nu1sance** [comment]
Thanks for your report greatly. The hardcoded channel values (512 and 256) in build_upsample_layer method (located in mmyolo/models/necks/re_cspnext_pafpn.py) cause a dimension mismatch when the widen_factor is changed.

To fix this, we need to dynamically scale the channels using self.widen_factor. Please update the build_upsample_layer method as follows:

```python
    def build_upsample_layer(self, idx: int, *args, **kwargs) -> nn.Module:
        """build upsample layer."""
        feat_2 = int(512 * self.widen_factor)
        feat_1 = int(256 * self.widen_factor)
        if idx == 2:
            return ennInterpolate(feat_2, **self.upsample_cfg)
        if idx == 1:
            return ennInterpolate(feat_1, **self.upsample_cfg)
        else:
            raise NotImplementedError
```

Let me know if this works for you\!

**Comment by wangwuwu** [comment]
> Thanks for your report greatly. The hardcoded channel values (512 and 256) in build_upsample_layer method (located in mmyolo/models/necks/re_cspnext_pafpn.py) cause a dimension mismatch when the widen_factor is changed.
> 
> To fix this, we need to dynamically scale the channels using self.widen_factor. Please update the build_upsample_layer method as follows:
> 
>     def build_upsample_layer(self, idx: int, *args, **kwargs) -> nn.Module:
>         """build upsample layer."""
>         feat_2 = int(512 * self.widen_factor)
>         feat_1 = int(256 * self.widen_factor)
>         if idx == 2:
>             return ennInterpolate(feat_2, **self.upsample_cfg)
>         if idx == 1:
>             return ennInterpolate(feat_1, **self.upsample_cfg)
>         else:
>             raise NotImplementedError
> Let me know if this works for you!

thank you , it works!
